mysqld_exporter.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"database/sql"
+	_ "github.com/go-sql-driver/mysql"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "mysql"
+)
+
+var (
+	listenAddress = flag.String("web.listen-address", ":9091", "Address to listen on for web interface and telemetry.")
+	metricPath    = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
+	configFile    = flag.String("config.file", "mysqld_exporter.conf", "Path to config file.")
+)
+
+type Config struct {
+	Config map[string]string `json:"config"`
+}
+
+type Exporter struct {
+	config                     Config
+	mutex                      sync.RWMutex
+	up                         prometheus.Gauge
+	totalScrapes, errorScrapes prometheus.Counter
+	metrics                    map[string]prometheus.Gauge
+}
+
+func getConfig(file string) (*Config, error) {
+	config := &Config{}
+
+	bytes, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	return config, json.Unmarshal(bytes, &config)
+}
+
+// return new empty exporter
+func NewMySQLExporter(config *Config) *Exporter {
+	return &Exporter{
+		config: *config,
+		up: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      "up",
+			Help:      "Was the last scrape of mysqld successful.",
+		}),
+		totalScrapes: prometheus.NewCounter(prometheus.CounterOpts{
```

Comment:
mysql_up should cover this and the next one already.

I'd suggest a gauge for how long the scrape of mysql took in seconds

http://prometheus.io/docs/practices/instrumentation/#collectors